rust/kernel/sync/nowait.rs

```diff
@@ -0,0 +1,188 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! A lock that never waits.
+
+use core::cell::UnsafeCell;
+use core::sync::atomic::{AtomicU8, Ordering};
+
+const LOCKED: u8 = 1;
+const CONTENDED: u8 = 2;
+
+/// A lock that only offers a [`NoWaitLock::try_lock`] method.
```

Comment:
Maybe only show `try_lock`, but still link to the full method?

Reply:
Yes, it looks better as you suggest.